packages/js/product-editor/src/components/header/header.tsx

```diff
@@ -1,14 +1,76 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { Product } from '@woocommerce/data';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { navigateTo, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { AUTO_DRAFT_NAME } from '../../utils';
 
 export type HeaderProps = {
-	title: string;
+	productId: number;
+	productName: string;
 };
 
-export function Header( { title }: HeaderProps ) {
+export function Header( { productId, productName }: HeaderProps ) {
```

Comment:
To not conflict with the useSelect, maybe we want to convert `productName` to `initialProductName` or something like that, or the one within the useSelect could be `editedProductName`.

Reply:
[1d0165e](https://github.com/woocommerce/woocommerce/pull/37152/commits/1d0165e875634df4d66ba20d715574f9e5423d29)